sim/pokemon.ts

```diff
@@ -537,11 +537,14 @@ export class Pokemon {
 				side.n % 2 === team ? side.active : []
 			);
 		}
-		if (adjacentOnly) allies = allies.filter(ally => this.battle.isAdjacent(this, ally));
 		return allies.filter(ally => ally && !ally.fainted);
 	}
 
-	foes(adjacentOnly?: boolean): Pokemon[] {
+	nearbyAllies(): Pokemon[] {
```

Comment:
This is going to conflict with the same change I commented on MacChaeger's PR...

Reply:
What's it? I can only find this comment, which seems to agree with my approach...

> I often wonder if these should be different functions, rather than the same function with different booleans passed to them. How much code do they even share?

See also: [https://ariya.io/2011/08/hall-of-api-shame-boolean-trap](Hall of API shame)

Reply:
Oh yes I agree. I'm just saying MacChaeger will have fun merging this.

Reply:
I did suggest this to MacChaeger before you even PRed this, after all. :p

I'm not sure how much I like "nearby" vs "adjacent", though.

Reply:
`adjacentAllies` etc seems like it would be better; to use the same word throughout the codebase.